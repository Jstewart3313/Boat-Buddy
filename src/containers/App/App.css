@import url('https://fonts.googleapis.com/css?family=Montserrat');

h1 {
  font-family: ‘Montserrat’, sans-serif;
  font-weight: 300;
}

header {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.active {
  color: #3376E8;
  text-decoration: underline;
}

.mission-link {
  margin-top: 20px;
  text-decoration: none;
}

.home-link {
  margin-top: 20px;
  text-decoration: none;
}

.App {
  background-color: #E5E8F4;
  margin: 0px 90px 0px 90px;
  text-align: center;
}


