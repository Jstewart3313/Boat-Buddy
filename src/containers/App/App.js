import React, { Component } from 'react';
import { Route, NavLink, Switch, Redirect } from 'react-router-dom';
import { mission } from '../../components/mission/mission'
import CardContainer from '../CardContainer/CardContainer';
import './App.css';

class App extends Component {
  render() {
    return (
      <main className="App">
      <header className='header'>
        <NavLink to='/' className='home-link'> Home</NavLink>
        <h1>BOAT BUDDY</h1>
        <NavLink to='/about' className='mission-link'>Mission </NavLink>
      </header>
      <Switch>
        <Route exact path='/about' component={ mission } />
        <Route exact path='/' component={ CardContainer } />
        <Redirect to={{pathname: '/'}} />
      </Switch>
      </main>
    );
  }
}

export default App;
