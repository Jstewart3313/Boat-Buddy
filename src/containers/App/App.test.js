import React from "react";
import { shallow } from "enzyme";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";

import App from "./App";

describe("App", () => {
  let mockStore;
  let initialState;
  let store;
  beforeEach(() => {
    mockStore = configureMockStore();
    initialState = {
      movies: [],
      music: [],
      tvShow: []
    };
    store = mockStore(initialState);
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<App store={store} />)
    expect(wrapper).toMatchSnapshot();
  });
});
