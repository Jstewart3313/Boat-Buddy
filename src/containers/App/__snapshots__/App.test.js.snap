// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<main
  className="App"
>
  <header
    className="header"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="home-link"
      to="/"
    >
       Home
    </NavLink>
    <h1>
      BOAT BUDDY
    </h1>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="mission-link"
      to="/about"
    >
      Mission 
    </NavLink>
  </header>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Redirect
      push={false}
      to={
        Object {
          "pathname": "/",
        }
      }
    />
  </Switch>
</main>
`;
